btax/calc_z.py

```diff
@@ -1,201 +1,301 @@
     tax_deprec_rates['System'] = tax_deprec_rates['GDS Life'].apply(str_modified)
-    tax_deprec_rates['System'].replace(deprec_system,inplace=True)
-    tax_deprec_rates.loc[tax_deprec_rates['System']=='ADS','Method'] = 'SL'
-    tax_deprec_rates.loc[tax_deprec_rates['System']=='Economic','Method'] = 'Economic'
+    tax_deprec_rates['System'].replace(deprec_system, inplace=True)
+    tax_deprec_rates.loc[tax_deprec_rates['System'] == 'ADS', 'Method'] = 'SL'
+    tax_deprec_rates.loc[tax_deprec_rates['System'] == 'Economic', 'Method'] = 'Economic'
 
     # add bonus depreciation to tax deprec parameters dataframe
     tax_deprec_rates['bonus'] = tax_deprec_rates['GDS Class Life'].apply(str_modified)
-    tax_deprec_rates['bonus'].replace(bonus_deprec,inplace=True)
+    tax_deprec_rates['bonus'].replace(bonus_deprec, inplace=True)
 
     # merge in econ depreciation rates
-    tax_deprec_rates = pd.merge(tax_deprec_rates, delta, how='left', left_on=['Asset Type'],
-      right_on=['Asset'], left_index=False, right_index=False, sort=False,
-      copy=True)
+    tax_deprec_rates = pd.merge(tax_deprec_rates, delta, how='left',
+                                left_on=['Asset Type'],
+                                right_on=['Asset'], left_index=False,
+                                right_index=False, sort=False, copy=True)
 
-    z = npv_tax_deprec(tax_deprec_rates, r, pi, tax_methods, financing_list, entity_list)
+    z = npv_tax_deprec(tax_deprec_rates, r, pi, tax_methods,
+                       financing_list, entity_list)
 
-    # replace tax depreciation rates on land and inventories w/ zero - unless expense
+    # replace tax depreciation rates on inventories w/ zero - unless expense
     if expense_inventory:
         for i in range(r.shape[0]):
             for j in range(r.shape[1]):
-                z.loc[z['Asset Type']=='Inventories', 'z'+entity_list[j]+financing_list[i]] = 1.
-    else:
-        for i in range(r.shape[0]):
-            for j in range(r.shape[1]):
-                z.loc[z['Asset Type']=='Inventories', 'z'+entity_list[j]+financing_list[i]] = 0.
-    if expense_land:
-        for i in range(r.shape[0]):
-            for j in range(r.shape[1]):
-                z.loc[z['Asset Type']=='Land', 'z'+entity_list[j]+financing_list[i]] = 1.
+                z.loc[z['Asset Type'] == 'Inventories', 'z' +
+                      entity_list[j] + financing_list[i]] = 1.
     else:
         for i in range(r.shape[0]):
             for j in range(r.shape[1]):
-                z.loc[z['Asset Type']=='Land', 'z'+entity_list[j]+financing_list[i]] = 0.
+                z.loc[z['Asset Type'] == 'Inventories',
+                      'z'+entity_list[j] + financing_list[i]] = 0.
+
+    # apply rate of expensing to land
+    for i in range(r.shape[0]):
+        for j in range(r.shape[1]):
+            z.loc[z['Asset Type'] == 'Land', 'z' + entity_list[j] +
+                  financing_list[i]] = expense_land
```

Comment:
@jdebacker I think it may be more efficient to build out the columns you want in the for loops and then set the values like:

```
for i in range(r.shape[0]):
    for j in range(r.shape[1]):
        columns_you_want.append('z' + entity_list[j] + financing_list[i]) 

z.loc[z['Asset Type'] == 'Land', columns_you_want] = expense_land
```

Reply:
@hdoupe Thank for this suggestion.  I'll incorporate this.